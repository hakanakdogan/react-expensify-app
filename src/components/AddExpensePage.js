import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { startAddExpense } from "../actions/expenses";
export class AddExpensePage extends React.Component { //connectle bağlanmamı versiyonu test etmek için export ettik
    onSubmit = (expense) => {
        //props.dispatch(addExpense(expense));
        this.props.startAddExpense(expense);
        this.props.history.push("/") //dashboard sayfasına değişecek
    }

    render() {
        return (
            <div>
                <h1>Add Expense </h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

// const AddExpensePage = (props) => (
//     <div>
//         <h1>Add Expense </h1>
//         <ExpenseForm
//             onSubmit={(expense) => {
//                 //props.dispatch(addExpense(expense));
//                 props.onSubmit(expense)
//                 props.history.push("/") //dashboard sayfasına değişecek
//             }}
//         />
//     </div>
// );

const mapDispatchToProps = (dispatch) => {
    return {
        startAddExpense: (expense) => {
            return dispatch(startAddExpense(expense))
        }
    }
}

export default connect(undefined, mapDispatchToProps)(AddExpensePage);