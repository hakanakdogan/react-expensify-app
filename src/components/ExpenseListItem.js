import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseListItem = ({ history, description, amount, createdAt, id, dispatch }) => (

    <div>
        <Link to={`/editexpense/${id}`}>
            <h3>{description}</h3>
        </Link>

        <p>
            {numeral(amount/100).format("$0,0.00")}
                -
            {moment(createdAt).format("MMMM Do, YYYY")}

        </p>






        {
            // <Link to={`/editexpense/${id}`}>
            //     <button>
            //         Edit
            //     </button>
            // </Link>

            //Butonu sayfa değiştirmek için böyle kullanabiliyorum
        }


        <hr />
    </div>

);


export default ExpenseListItem;