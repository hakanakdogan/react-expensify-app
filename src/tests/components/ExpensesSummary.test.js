import React from "react";
import { shallow } from "enzyme";
import {ExpensesSummary} from "../../components/ExpensesSummary";

test("Should render ExpensesSummary Correctly with 1 expense", () => {
  const wrapper = shallow(<ExpensesSummary expenseCount = {1} expensesTotal = {10} />);
  expect(wrapper).toMatchSnapshot();
})

test("Should render ExpensesSummary Correctly with multiple expenses", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount = {10} expensesTotal = {100} />);
    expect(wrapper).toMatchSnapshot();
  })