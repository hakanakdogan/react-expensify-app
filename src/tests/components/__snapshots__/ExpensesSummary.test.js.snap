// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary Correctly with 1 expense 1`] = `
<div>
  <h4>
    Viewing 
    1
     
    expense
     totalling 
    $0.10
     
  </h4>
</div>
`;

exports[`Should render ExpensesSummary Correctly with multiple expenses 1`] = `
<div>
  <h4>
    Viewing 
    10
     
    expenses
     totalling 
    $1.00
     
  </h4>
</div>
`;
